src/Microsoft.Health.Dicom.Core/Features/ExtendedQueryTag/AddExtendedQueryTagErrorsService.cs

```diff
@@ -0,0 +1,32 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+
+namespace Microsoft.Health.Dicom.Core.Features.ExtendedQueryTag
+{
+    public class AddExtendedQueryTagErrorsService : IAddExtendedQueryTagErrorsService
+    {
+        private readonly IExtendedQueryTagErrorStore _extendedQueryTagErrorStore;
+
+        public AddExtendedQueryTagErrorsService(IExtendedQueryTagErrorStore extendedQueryTagErrorStore)
+        {
+            _extendedQueryTagErrorStore = EnsureArg.IsNotNull(extendedQueryTagErrorStore, nameof(extendedQueryTagErrorStore));
+        }
+
+        public async Task<int> AddExtendedQueryTagErrorAsync(int tagKey, int errorCode, long watermark, DateTime createdTime, CancellationToken cancellationToken = default)
+        {
+            return await _extendedQueryTagErrorStore.AddExtendedQueryTagErrorAsync(
```

Comment:
nit: return the `Task` directly.

Furthermore, I don't think we need this service at the end of the day per operation, like I think the extended query tags have. Instead, I think it is sufficient for the handlers to exist per operation, but a single service can encapsulate all of the operations on the store.